common/src/main/java/com/wynntils/screens/skillpointloadouts/SkillPointLoadoutsScreen.java

```diff
@@ -29,17 +30,22 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
 import net.minecraft.ChatFormatting;
 import net.minecraft.client.gui.GuiGraphics;
 import net.minecraft.client.gui.components.Tooltip;
 import net.minecraft.client.gui.screens.Screen;
 import net.minecraft.client.resources.language.I18n;
 import net.minecraft.network.chat.Component;
+import org.lwjgl.glfw.GLFW;
 
 public final class SkillPointLoadoutsScreen extends WynntilsGridLayoutScreen {
+    private static final int MAX_LOADOUTS_PER_PAGE = 11;
     private List<LoadoutWidget> loadoutWidgets = new ArrayList<>();
 
     private boolean firstInit = true;
+    private final List<Pair<Supplier<String>, Function<Skill, Integer>>> summaryParts = new ArrayList<>();
```

Comment:
Why supplier? Also if it ends up as a StyledText, why save it as StyledText?

Reply:
Needs to be supplier since the skill point sums used in the I18n %s parts can change when the screen is still open. If they aren't suppliers they will get stuck on the first parse result

Reply:
And why not make a method for repopulating a list of cached strings when they change? I think that should be a bit easier logic to have overall.

Reply:
I don't think that is much better than what the suppliers do now. If we do the cache and repopulation, we need to get the model to post an event, have the feature subscribe, and call some public method in the screen to repopulate cache. But the supplier way is essentially the exact same as if you had manually coded all the summary lines into doRender(), and I don't think it's expensive to just get a number from a model.

Reply:
Okay, let's do this (for now, at least).